scaleway/validation.go

```diff
@@ -0,0 +1,68 @@
+package scaleway
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+// validationUUID validate the schema follows the canonical UUID format
+// "6ba7b810-9dad-11d1-80b4-00c04fd430c8".
+func validationUUID() func(interface{}, string) ([]string, []error) {
+	return func(v interface{}, key string) (warnings []string, errors []error) {
+		uuid, isString := v.(string)
+		if !isString {
+			return nil, []error{fmt.Errorf("invalid UUID: not a string")}
+		}
+
+		t := []byte(uuid)
+		if len(t) != 36 || t[8] != '-' || t[13] != '-' || t[18] != '-' || t[23] != '-' {
+			return nil, []error{fmt.Errorf("invalid UUID '%s' (%d): format should be 'xxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' (36)", uuid, len(uuid))}
+		}
+
+		_, err := hex.DecodeString(strings.Replace(uuid, "-", "", -1))
+		if err != nil {
+			return nil, []error{fmt.Errorf("invalid UUID '%s': characters should be valid hexadecimal", uuid)}
+		}
+
+		return
+	}
+}
+
+func validationZone() func(interface{}, string) ([]string, []error) {
+	return func(v interface{}, key string) (warnings []string, errors []error) {
+		rawZone, isString := v.(string)
+		if !isString {
+			return nil, []error{fmt.Errorf("invalid zone: not a string")}
+		}
+
+		zone, _ := utils.ParseZone(rawZone)
+		if rawZone == "par1" || rawZone == "ams1" {
+			warnings = append(warnings, fmt.Sprintf("%s is a deprecated name for zone, use %v instead", rawZone, zone))
+		} else if !zone.Exists() {
+			warnings = append(warnings, "%s zone does not exist", rawZone)
+		}
+
+		return
+	}
+}
+
+func validationRegion() func(interface{}, string) ([]string, []error) {
+	return func(v interface{}, key string) (warnings []string, errors []error) {
+		rawRegion, isString := v.(string)
+		if !isString {
+			return nil, []error{fmt.Errorf("invalid region: not a string")}
+		}
+
+		region, _ := utils.ParseRegion(rawRegion)
```

Comment:
Add a `// TODO: Use utils.ParseRegion when the format validation will be implemented.`?